grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -36,6 +35,7 @@ class InvoiceItem implements Serializable {
 
     String id
     Invoice invoice
+    InvoiceItemType invoiceItemType
```

Comment:
@jmiranda let me know if you prefer this one as a separate `InvoiceItemType` domain that has `name` and `code` as  `InvoiceItemTypeCode` enum instead straight away using enum here, and I'll adjust that

Reply:
Yeah, now that I'm seeing how it's being used, I don't think InvoiceItemType is correct. An invoice item type would describe what the invoice item is for i.e. product, service, adjustment, subscription, labor, tax, discount, fees, etc.

What we're talking about here is something I still don't really understand. But it feels like it might be the invoice type (prepayment, standard, final) and should go at the Invoice level. 

This is why I wish we had done the "from scratch" design instead of trying to build on what we had (at least from the data model perspective) so that we could have spent more time thinking about the proper way to do this. Frankly, I don't have a good enough understanding of it so I guess I'll leave it up to you. But I would really consider moving this "type" to the invoice and derive the "inverse" logic based on, I guess whether the item is in an invoice with InvoiceTypeCode.FINAL (and whatever other)

In either case, the safer option would be to use the enum here since we could always migrate to using a type domain that maps to the enum later.